### PR TITLE
feat: #337 add options to ASelect to handle long menu items

### LINKED ISSUE #337: Consider limiting the Select option text
In certain cases the options we display for dropdown menus are coming from user input and we can end up in a situation where the width of the option goes beyond the screen viewport:

Screenshot 2023-04-03 at 15 38 40

Also, if the user were to select this long option - we may want to truncate the text so it fits on 1 line only
Screenshot 2023-04-03 at 15 40 37

Code to reproduce:
```
() => {
  const items = [
    {id: 1, name: "Bread, Cereal, Rice, and Pasta"},
    {id: 2, name: "Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables Vegetables All the way"}
  ];
  const [selectedItem, setSelectedItem] = useState(items[2]);
  return (
    <div style={{minHeight: 220, minWidth: 250, marginRight: 20}}>
      <ASelect
        label="Food Group"
        items={items}
        itemText="name"
        itemValue="id"
        placeholder="Pick a Food Group"
        onSelected={(item) => setSelectedItem(item)}
        hint={`Selected Item: ${JSON.stringify(selectedItem)}`}
      />
    </div>
  );
}
```

### PR DESCRIPTION
**Pull request by rwharrisjr**
Adds two props to ASelect

textWrapMenuItems | If item display is a string (no template), set white-space:normal on each item and limit width of the menu.Boolean | null
-- | -- | --
truncateMenuItems | If item display is a string (no template), truncate the display string with ellipsis and limit width of the menu.Boolean | null

Resolves #337

### CODE REVIEW

File: framework/components/AMenu/AMenu.js [hunk 1]
```diff
@@ -96,7 +96,7 @@ const AMenu = forwardRef(
     if (propsClassName) {
       className += ` ${propsClassName}`;
     }
-
+    console.log(rest);
```
Comment (vdineva) [inline framework/components/AMenu/AMenu.js:99]: you'd probably want to remove the `console.log` before merging it in

File: framework/components/AMenuBase/AMenuBase.js [hunk 2]
```diff
@@ -356,6 +362,7 @@ const AMenuBase = forwardRef(
       pointerStyle.top = pointerTop;
     }
 
+    console.log("style", style);
```
Comment (vdineva) [inline framework/components/AMenuBase/AMenuBase.js:365]: you'd probably want to remove the `console.log` before merging it in

**Review by vdineva — APPROVED** [review]
lgtm- with minor comments

File: framework/components/AMenuBase/AMenuBase.js [hunk 3]
```diff
@@ -24,7 +24,9 @@ const calculateMenuPosition = (
   setInvisible,
   removeSpacer = false
 ) => {
-  if (!combinedRef.current) return;
```
Comment (brennarvo) [inline framework/components/AMenuBase/AMenuBase.js:27]: 👏

**Comment by brennarvo** [comment]
I made two small updates:

* Adds the `title` attribute for when the select options are truncated so that the user is aware of what they are selecting
* Preserves the `text-indent` on select options whose text wraps onto a new line

**Pull request merged**